service/src/main/kotlin/io/provenance/api/frameworks/provenance/extensions/ProvenanceExtensions.kt

```diff
@@ -99,9 +99,9 @@ fun Any.toFeegrantAllowance(): Allowance? =
         this.typeUrl.endsWith("v1beta1.AllowedMsgAllowance") ->
             this.unpack(Feegrant.AllowedMsgAllowance::class.java)
                 .let {
-                    FeeGrantAllowedMsgAllowance(it.allowance.toFeegrantAllowance()!!, it.allowedMessagesList)
+                    FeeGrantAllowedMsgAllowance(it.allowance.toFeeGrantAllowance(), it.allowedMessagesList)
                 }
-        else -> null.also { logger.error("Invalid feegrant type: ${this.typeUrl}") }
+        else -> throw IllegalStateException("No valid allowances exist for specified fee grant.")
```

Comment:
Didn't want to keep the type URL logging in the exception message?

Reply:
LOL just felt the error was specific enough since there's a known amount of valid types and can always be inspected elsewhere. I'll just add the original error message back:P

Reply:
You're right, sorry for all the nits 😅